ui/js/signup.js

```diff
@@ -319,6 +319,53 @@ require(['jquery', 'underscore', 'oae.core'], function($, _, oae) {
     };
 
     /**
+     * Initialize the signup institution search template
+     */
+    var renderSignUpInstitutionSearch = function() {
+        if (!oae.data.me.tenant.isGuestTenant) {
+            // Don't render institution search at all if we're not on the guest
+            // tenant
+            return;
+        }
+
+        oae.api.util.template().render($('#signup-institution-template'),
+            null, $('#signup-institution-container'));
+
+        oae.api.util.autoSuggest().setup($('#signup-institution-search-field'), {
+            'url': '/api/search/tenants',
+            'formatList': function(data, $el) {
+                var countryInfo = _.findWhere(iso3166.countries, {'code': data.countryCode});
+                oae.api.util.template().render($('#signup-autosuggest-tenant-template'), {
+                    'tenant': _.extend({}, data, {
+                        'countryName': (countryInfo && countryInfo.name),
+                        'countryFlag': (countryInfo && countryInfo.icon)
+                    })
+                }, $el);
+                return $el;
+            },
+            'selectionAdded': function(el) {
```

Comment:
I think this interaction needs a lot more direction to avoid wasting too much development time on it.

What does it look like when you've selected an institution, what's the state of the field?
How do you "back out" and search for a different institution?
What happens to the state of the field and the "Go" button when you've backed out?